model/Infrastructure/QtiTestRepository.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoQtiTest\model\Infrastructure;
+
+use core_kernel_classes_Resource;
+use oat\generis\model\data\Ontology;
+use oat\taoDeliveryRdf\model\DeliveryAssemblyService;
+use oat\taoQtiItem\model\qti\Service;
+use oat\taoQtiTest\model\Domain\Model\QtiTest;
+use oat\taoQtiTest\model\Domain\Model\QtiTestRepositoryInterface;
+use oat\taoQtiTest\models\TestModelService;
+
+class QtiTestRepository implements QtiTestRepositoryInterface
+{
+    /** @var Ontology */
+    private $ontology;
+
+    /** @var TestModelService */
+    private $testModelService;
+
+    /** @var Service */
+    private $qtiItemService;
+
+    public function __construct(Ontology $ontology, TestModelService $testModelService, Service $qtiItemService = null)
+    {
+        $this->ontology = $ontology;
+        $this->testModelService = $testModelService;
+        $this->qtiItemService = $qtiItemService ?? Service::singleton();
```

Comment:
[Optional]
You can register this service in DI container via factory and provide it to constructor as a service. Example:
* https://github.com/oat-sa/tao-core/blob/master/models/classes/accessControl/AccessControlServiceProvider.php#L43
* https://github.com/oat-sa/tao-core/blob/master/models/classes/accessControl/AccessControlServiceProvider.php#L50